applications/IgaApplication/custom_modelers/nurbs_geometry_modeler_sbm.cpp

```diff
@@ -0,0 +1,126 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+// Project includes
+#include "nurbs_geometry_modeler_sbm.h"
+#include "custom_utilities/create_breps_sbm_utilities.h"
+#include "utilities/nurbs_utilities/snake_sbm_utilities.h"
+
+namespace Kratos
+{
+
+    ///@name Stages
+    ///@{
+
+    void NurbsGeometryModelerSbm::SetupGeometryModel(){
+
+        //----------------------------------------------------------------------------------------------------------------
+        KRATOS_INFO_IF("NurbsGeometryModelerSbm", mEchoLevel > 1) << "[NURBS MODELER SBM]:: calling NurbsGeometryModeler" << std::endl;
+        
+        // Call the SetupGeometryModel method of the base class NurbsGeometryModeler
+        NurbsGeometryModeler::SetupGeometryModel();
+    }
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+    void NurbsGeometryModelerSbm::CreateAndAddRegularGrid2D( ModelPart& r_model_part, const Point& A_xyz, const Point& B_xyz,
+        const Point& A_uvw, const Point& B_uvw, SizeType OrderU, SizeType OrderV,SizeType NumKnotSpansU, SizeType NumKnotSpansV, bool add_surface_to_model_part)
```

Comment:
I assume this is an `override` of a `virtual` member. If so, mark it as such. Also, you can use the `@copydoc` doxygen command to pull in the annotations from the base class.

```suggestion
    
    /// @copydoc NurbsGeometryModeler::CreateAndAddRegularGrid2D
    void NurbsGeometryModelerSbm::CreateAndAddRegularGrid2D( ModelPart& r_model_part, const Point& A_xyz, const Point& B_xyz,
        const Point& A_uvw, const Point& B_uvw, SizeType OrderU, SizeType OrderV,SizeType NumKnotSpansU, SizeType NumKnotSpansV, bool add_surface_to_model_part) override
```

 \+ same comment about variable naming in general.